scripts/dive_compositor_capture.sh

```diff
@@ -18,62 +18,86 @@ set -eux
-echo -n "Waiting for surfaceflinger to restart... "
-for i in $(seq ${TIME_TO_WAIT_FOR_RESTART} -1 1); do
-    echo -n "$i, "
-    sleep 1
-done
-echo "done."
+  adb shell setprop cpm.gfxr_layer 1
+  adb shell stop && adb shell start
+  # TODO(b/449739284): find a better way to detect surfaceflinger done initialization.
+  echo -n "Waiting for surfaceflinger to restart... "
+  for i in $(seq ${TIME_TO_WAIT_FOR_RESTART} -1 1); do
+      echo -n "$i, "
+      sleep 1
+  done
+  echo "done."
+}
 
-# Copied from the gfxr_capture.sh script.
-while true; do
-  # Let the user dictate which frame they want the capture.
-  echo "Press key g+enter to trigger a capture and g+enter to retrieve the capture."
-  echo "Press any other key+enter to stop the application."
-  read reply
-  if [ "${reply}" != "g" ]; then
-    break
-  fi
+trigger_capture() {
+  # Copied from the gfxr_capture.sh script.
+  while true; do
+    # Let the user dictate which frame they want the capture.
+    echo "Press key g+enter to trigger a capture and g+enter to retrieve the capture."
+    echo "Press any other key+enter to stop the application."
+    read reply
+    if [ "${reply}" != "g" ]; then
+      break
+    fi
 
-  adb shell setprop debug.gfxrecon.capture_android_trigger true
+    adb shell setprop debug.gfxrecon.capture_android_trigger true
 
-  # Don't know when the capture has finished so defer to the user.
-  # Ideally, they have `adb logcat -s gfxrecon` running in another terminal.
-  echo "Press any key to retrieve capture"
-  read
-  adb shell setprop debug.gfxrecon.capture_android_trigger false
-  # Easiest to just pull the entire directory.
-  adb pull ${COMPOSITOR_CAPTURE_DIR} ./captures
-  # Clean up so that we're not constantly pulling the same files.
-  # Also causes names to be printed to output so the user knows.
-  for file in $(adb shell ls "${COMPOSITOR_CAPTURE_DIR}"); do  
-    adb shell rm "${COMPOSITOR_CAPTURE_DIR}/${file}"
+    # Don't know when the capture has finished so defer to the user.
+    # Ideally, they have `adb logcat -s gfxrecon` running in another terminal.
+    echo "Press any key to retrieve capture"
+    read
+    adb shell setprop debug.gfxrecon.capture_android_trigger false
+    # Easiest to just pull the entire directory.
+    adb pull ${COMPOSITOR_CAPTURE_DIR} ./captures
+    # Clean up so that we're not constantly pulling the same files.
+    # Also causes names to be printed to output so the user knows.
+    for file in $(adb shell ls "${COMPOSITOR_CAPTURE_DIR}" | tr -d '\r'); do
+      adb shell rm "${COMPOSITOR_CAPTURE_DIR}/${file}"
+    done
   done
-done
+}
+
+cleanup() {
+  adb shell rm -rf ${COMPOSITOR_CAPTURE_DIR}
+  adb shell rm /data/local/debug/vulkan/libVkLayer_gfxreconstruct.so
+  adb shell setprop cpm.gfxr_layer 0
+  adb shell setprop persist.compositor.protected_context "\"\""
+  adb shell setprop persist.compositor.tiled_rendering "\"\""
+  adb shell setprop compositor.lazy_depth_buffer "\"\""
```

Comment:
optional: you can avoid escaping by combining single and double quotes. I think it looks cleaner but that's personal choice. I can see what it may not be more readable since the reader could get visually confused with the number of quotes
```suggestion
  adb shell setprop persist.compositor.protected_context '""'
  adb shell setprop persist.compositor.tiled_rendering '""'
  adb shell setprop compositor.lazy_depth_buffer '""'
```